client/my-sites/sharing/connections/service.jsx

```diff
@@ -1,128 +1,182 @@
+import { getKeyringConnectionsByName, getAvailableExternalConnections } from 'state/sharing/keyring/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteUserConnectionsForService, isFetchingConnections } from 'state/sharing/publicize/selectors';
+import observe from 'lib/mixins/data-observe';
+import PopupMonitor from 'lib/popup-monitor';
+import { recordGoogleEvent } from 'state/analytics/actions';
+import ServiceAction from './service-action';
+import ServiceConnectedAccounts from './service-connected-accounts';
+import ServiceDescription from './service-description';
+import ServiceExamples from './service-examples';
+import services from './services';
+import ServiceTip from './service-tip';
+import SocialLogo from 'components/social-logo';
+
+const SharingService = React.createClass( {
 	displayName: 'SharingService',
 
 	propTypes: {
 		site: React.PropTypes.object,                    // The site for which connections are created
 		user: React.PropTypes.object,                    // A user object
 		service: React.PropTypes.object.isRequired,      // The single service object
-		connections: React.PropTypes.object.isRequired,  // A collections-list instance
-		onAddConnection: React.PropTypes.func,           // Handler for creating a new connection for this service
-		onRemoveConnection: React.PropTypes.func,        // Handler for removing a connection from this service
-		onRefreshConnection: React.PropTypes.func,       // Handler for refreshing a Keyring connection for this service
-		onToggleSitewideConnection: React.PropTypes.func // Handler to invoke when toggling a connection to be shared sitewide
+		connections: React.PropTypes.object,             // A collections-list instance
 	},
 
 	mixins: [ observe( 'connections' ) ],
 
-	getInitialState: function() {
-		return {
-			isOpen: false,          // The service is visually opened
-			isConnecting: false,    // A pending connection is awaiting authorization
-			isDisconnecting: false, // A pending disconnection is awaiting completion
-			isRefreshing: false     // A pending refresh is awaiting completion
-		};
+	/**
+	 * Returns the available connections for the current user.
+	 *
+	 * @return {Array} Available connections.
+	 */
+	getConnections: function() {
+		return this.filter( 'getConnections', this.props.service.ID, this.props.siteUserConnections, arguments );
 	},
 
-	getDefaultProps: function() {
-		return {
-			onAddConnection: function() {},
-			onRemoveConnection: function() {},
-			onRefreshConnection: function() {},
-			onToggleSitewideConnection: function() {}
-		};
+	/**
+	 * Given a service name, returns the connections for which the current user is permitted to remove.
+	 *
+	 * @param {string} service The name of the service
+	 * @return {Array} Connections for which the current user is permitted to remove
+	 */
+	getRemovableConnections: function( service ) {
```

Comment:
Should these be state selectors instead? I also regret the `filter` approach, maybe instead just a `switch` on the service ID in the selector?